### PR TITLE
auto fill add address

### PR DESCRIPTION
**Pull request by EagleEye25**
added functionality to add address to address book without the dialog initially being opened

### CODE REVIEW

**Comment by pciavald** [comment]
I found an unrelated bug when reviewing : adding my own address, the address appears in red and the `add label` button is disabled, but clicking cancel triggers the error that should have been triggered by the address checking. Only ESC key allows to get out of this situation. Can you fix that ?

screenshot_20171129_142351

**Comment by pciavald** [comment]
Latest restructure seems to pass the Travis tests, maybe rebase by adding a git remote on your repo, and doing `git pull upstream restructure`

**Comment by pciavald** [comment]
By looking at the commits here it seems there is a problem in the branches. Can you please rebase to the latest restructure and push ? There should be only your commits.

File: src/app/wallet/shared/util/utils.ts [hunk 1]
```diff
@@ -145,3 +145,21 @@ export class Duration {
   }
 
   }
+
+export class AddressHelper {
+  addressRegex: RegExp = /^[pPrRTt][a-km-zA-HJ-NP-Z1-9]{25,}$/;
```
Comment (kewde) [inline src/app/wallet/shared/util/utils.ts]: Does this work on both testnet and mainnet?

File: src/app/wallet/wallet/address-book/modal/new-address-modal/new-address-modal.component.html [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+    <md-form-field>
+      <input mdInput type="text" formControlName="address"
+             [ngModel]="address" (ngModelChange)="verifyAddress()"
              placeholder="Enter Address"
              [ngClass]="{'verify-sucess': checkAddress() == true, 'verify-error': checkAddress() == false }"
-             (input)="verifyAddress()"
              #addressInput
              required>
-    </mat-form-field>
+    </md-form-field>
```
Comment (kewde) [inline src/app/wallet/wallet/address-book/modal/new-address-modal/new-address-modal.component.html]: md-form-field? Shouldn't be possible...

File: src/app/wallet/shared/util/utils.ts (hunk 1 above)
Comment (EagleEye25) [reply src/app/wallet/shared/util/utils.ts]: yes it works on both, just tested it now

File: src/app/wallet/wallet/address-book/modal/new-address-modal/new-address-modal.component.html (hunk 2 above)
Comment (EagleEye25) [reply src/app/wallet/wallet/address-book/modal/new-address-modal/new-address-modal.component.html]: fixed

**Comment by pciavald** [comment]
seems to be working accordingly to what it should. There's still a problem when adding my own address in the address book and canceling but it should be fixed by #434.

**Pull request merged**